ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -210,13 +209,16 @@ PullQueryResult executeTablePullQuery(
       }
       return result;
     } catch (final Exception e) {
-      final PullSourceType finalSourceType = sourceType;
-      final PullPhysicalPlanType finalPlanType = planType;
-      final RoutingNodeType finalRoutingNodeType = routingNodeType;
-      pullQueryMetrics.ifPresent(metrics -> metrics.recordErrorRate(1,
-          Optional.ofNullable(finalSourceType).orElse(PullSourceType.UNKNOWN),
-          Optional.ofNullable(finalPlanType).orElse(PullPhysicalPlanType.UNKNOWN),
-          Optional.ofNullable(finalRoutingNodeType).orElse(RoutingNodeType.UNKNOWN)));
```

Comment:
The main purpose of this PR was to get rid of the XYType `UNKNOWN` values because they only exist for statements like these. The thing to note is that they are either all known or none known, so we don't need a per-field null-check and a per-type "unknown" element.

As a newcomer to the code base, I found myself really confused as to how the could possibly not know the type for a query. The answer is if the planning fails for the query, then there is no query and therefore no known type. However, if planning fails for the query, we also don't know anything else about it, so it's more straightforward to reason about it that way than on a per-field basis.